src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,119 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elementData = new Object[DEFAULT_CAPACITY];
+    private int size;
+
     @Override
     public void add(T value) {
-
+        for (int i = 0; i < size; i++) {
+            if (elementData.length == size) {
+                elementData = Arrays.copyOf(elementData,
```

Comment:
Use System.arraycopy() to move your array elements.
https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/array-list.html